app/scripts/metamask-controller.js

```diff
@@ -342,20 +341,14 @@ module.exports = class MetamaskController extends EventEmitter {
       version,
       // account mgmt
       getAccounts: async ({ origin }) => {
-        // Expose no accounts if this origin has not been approved, preventing
-        // account-requring RPC methods from completing successfully
-        const exposeAccounts = this.providerApprovalController.shouldExposeAccounts(origin)
-        if (origin !== 'metamask' && !exposeAccounts) {
-          return []
-        }
-        const isUnlocked = this.keyringController.memStore.getState().isUnlocked
-        const selectedAddress = this.preferencesController.getSelectedAddress()
-        // only show address if account is unlocked
-        if (isUnlocked && selectedAddress) {
-          return [selectedAddress]
-        } else {
-          return []
+        if (origin === 'metamask') {
+          return [this.preferencesController.getSelectedAddress()]
```

Comment:
Is there any chance this could return an array of length 1, where the one element is `undefined` or an empty string? if so, this will return a different value from before

Reply:
Fixed in 0950b7d